### PR TITLE
l2geth: fix startup logic for repairing queue index

### PR DESCRIPTION
**Pull request by tynes**
**Description**
Fix a comparison on startup

### CODE REVIEW

File: l2geth/rollup/sync_service.go [hunk 1]
```diff
@@ -348,10 +348,10 @@ func (s *SyncService) initializeLatestL1(ctcDeployHeight *big.Int) error {
 			qi := tx.GetMeta().QueueIndex
 			// When the queue index is set
 			if qi != nil {
-				if qi == queueIndex {
+				if *qi == *queueIndex {
 					log.Info("Found correct staring queue index", "queue-index", qi)
```
Comment (Inphi) [inline l2geth/rollup/sync_service.go:352]: can also deref `qi` here.

Comment (tynes) [reply l2geth/rollup/sync_service.go]: Good catch, fixing

**Review by Inphi — APPROVED** [review]

**Comment by smartcontracts** [comment]
Plz do not merge this PR until I perform a quick release, it messes up the release process

**Review by cfromknecht — APPROVED** [review]
LGTM, what are the implications here?

**Comment by tynes** [comment]
The implications here are that the original logic for fixing the queue was broken, it would always hit the else clause since they would never match in the initial if clause. This exists to prevent deposits from being skipped/double ingested on restarts when the deposit is partway through the ingestion pipeline

**Pull request merged**